File: sample_source_code/csharp/src/IAnimal.cs
namespace Embold.Sample;

/// <summary>Interface defining the contract for all animals.</summary>
public interface IAnimal
{
    string Name    { get; }
    string Species { get; }
    int    Age     { get; }
    double Weight  { get; }

    string Sound();
    string Describe();

    /// <summary>Accept a visitor — enables double-dispatch.</summary>
    void Accept(IVisitor visitor);
}

File: sample_source_code/typescript/src/IAnimal.ts
import type { IVisitor } from './IVisitor';

/** Interface defining the contract for all animals. */
export interface IAnimal {
    readonly name:    string;
    readonly species: string;
    readonly age:     number;
    readonly weight:  number;
    sound():   string;
    describe(): string;
    /** Accept a visitor — enables typed double-dispatch. */
    accept(visitor: IVisitor): void;
}
